### ISSUE TITLE
Visualize mesh normals

### ISSUE THREAD

**Issue by ndrplz** [IntelVCL/Open3D#765]
I'm a bit struggling to understand how to get to a visualization similar to [this one](http://open3d.org/docs/_images/normal.png) but for a mesh instead of a point cloud.

Currently what I'm doing is the following:
```python
mesh = read_triangle_mesh("knot.ply")
mesh.compute_vertex_normals()
mesh.vertex_colors = mesh.vertex_normals
draw_geometries([mesh])
```
which outputs the following:
![image](https://user-images.githubusercontent.com/3187598/50840001-d5215e00-1361-11e9-9845-e1a9f054f2cd.png)

Is it correct / is there a more principled way to do it?

Thank you.
Andrea

**Comment by qianyizh** [comment]
The easiest way to mimic point cloud normal rendering behavior is to press ``ctrl+9`` to enable normal rendering for mesh (for point cloud it is ``9``).
![screencapture_2019-01-08-10-34-20](https://user-images.githubusercontent.com/5875217/50851237-151c1d00-1331-11e9-9f33-b91132e7eb6a.png)

**Comment by ndrplz** [comment]
Thanks, that's what I meant. However, I'd like to toggle this mode automatically.

Currently as a workaround I'm using
```python
opt = vis.get_render_option().load_from_json("renderoption.json")
```
where in `renderoption.json` among other options I have
```json
 `mesh_color_option` : 9  # normal rendering
```
(for future readers, mesh color options are [here](https://github.com/IntelVCL/Open3D/blob/844be751ee76a0b2e5c20c4bf19d8e49ecbdb824/src/Visualization/Visualizer/RenderOption.h#L69-L76)).

Still, I tried to do something like 
```python
    opt = vis.get_render_option()
    opt.mesh_color_option = 9
```
but it doesn't seem to like it.

**Comment by qianyizh** [comment]
Yeah, it is not exposed right now. The current workaround is to use ``load_from_json``.
In the next restructure (hopefully coming soon) we will expose as many APIs in Python as possible.

**Comment by ndrplz** [comment]
Re-opening the issue for a strictly related thing. If I set the `mesh_color option` in the json to either:

 `mesh_color_option` : 2
 `mesh_color_option` : 3
 `mesh_color_option` : 4

I expect to see the rendering of X, Y and Z heatmaps respectively (analogously to what happens with ctrl+2/3/4). However this doesn't happen. Do I miss something?

**Comment by syncle** [comment]
Thanks for the feedback @ndrplz. Let me work on this.

**Comment by syncle** [comment]
I just pushed new PR for this.

The following code
```
from open3d import *

geometry = read_triangle_mesh("TestData/knot.ply")
geometry.compute_vertex_normals()
print(geometry)

vis = Visualizer()
vis.create_window()
vis.add_geometry(geometry)
vis.get_render_option().mesh_color_option_ = MeshColorOption.Normal
vis.run()
vis.destroy_window()
```
should produce this:
<img width="1072" alt="screen shot 2019-01-16 at 1 54 49 pm" src="https://user-images.githubusercontent.com/18297113/51281022-1800b880-1996-11e9-8deb-095de76b1635.png">

**Comment by syncle** [comment]
FYI, this fix will be shipped with 0.5 release.
